modin/engines/base/frame/data.py

```diff
@@ -1027,13 +1026,13 @@ def merge(left_index, right_index):
                 # TODO: revisit for performance
                 for index in indexes[1:]:
                     joined_index = merge(joined_index, index)
-
-                if is_duplicates:
-                    for i, index in enumerate(indexes):
-                        indexers[i] = index.get_indexer_for(joined_index)
         else:
             joined_index = indexes[0].copy()
 
+        if is_duplicates and indexers[0] is None:
+            for i, index in enumerate(indexes):
+                indexers[i] = index.get_indexer_for(joined_index)
```

Comment:
Why moved?

Reply:
if `do_join_index is False` but `is_duplicates is True` we want to get reindexers, so moved it under `if do_join_index` branch to be available in cases when we don't want to join indices

Reply:
Why do we need renindexers if we don't do join?

Reply:
it happens when `how` parameter is `None`, in that case we're not joining indexes by considering `joined_index` as `self.index`

Reply:
So more clear condition is `if is_duplicates and how is None`, isn't it?

Reply:
Also note that in case when:
```python
any(not indexes[0].equals(index) for index in [indexes[1:]]) == True
```
we don't need to get indexers

Reply:
>So more clear condition is if is_duplicates and how is None, isn't it?

this won't work in cases when `how is not None`. 
Why do we have a statement like that:
We want to get `indexers` when `is_duplicated and (do_join_index or how is not None)`, when `do_join_index is True` we're falling into [this branch](https://github.com/dchigarev/modin/blob/1c6b2588318960381d467d704ba0392abe05b1f4/modin/engines/base/frame/data.py#L1016) and if `len(indexes) == 2 and is_duplicates` we're computing `indexers` in a fast path, so don't want to compute them again, otherwise we should compute them further. After determining what `joined_index` is, we're hitting [this if](https://github.com/dchigarev/modin/blob/1c6b2588318960381d467d704ba0392abe05b1f4/modin/engines/base/frame/data.py#L1032) that says, that if we need indexers (`is_duplicates is True`) but didn't computed it earlier (`indexers[0] is None`) them compute indexers

Reply:
reworked it in the last commit